core/src/parse.rs

```diff
@@ -442,10 +463,46 @@ mod tests {
 
     #[test]
     fn test_whitespace() {
-        assert_eq!(parse("").unwrap().body, vec![]);
-        assert_eq!(parse("\n\n\n").unwrap().body, vec![]);
-        assert_eq!(parse("  ").unwrap().body, vec![]);
-        assert_eq!(parse("  \n  ").unwrap().body, vec![]);
+        assert_eq!(
+            parse("LBL:\n\n\n").unwrap().body,
+            vec![Node(
+                Statement::Label(Node(
+                    LabelDecl("LBL".into()),
+                    span(1, 1, 1, 5)
+                )),
+                span(1, 1, 1, 5)
+            )]
+        );
+        assert_eq!(
+            parse("\n\nLBL:\n\n").unwrap().body,
+            vec![Node(
+                Statement::Label(Node(
+                    LabelDecl("LBL".into()),
+                    span(3, 1, 3, 5)
+                )),
+                span(3, 1, 3, 5)
+            )]
+        );
+        assert_eq!(
+            parse("  LBL:").unwrap().body,
+            vec![Node(
+                Statement::Label(Node(
+                    LabelDecl("LBL".into()),
+                    span(1, 3, 1, 7)
+                )),
+                span(1, 3, 1, 7)
+            )]
+        );
+        assert_eq!(
+            parse("  \n  LBL:").unwrap().body,
+            vec![Node(
+                Statement::Label(Node(
+                    LabelDecl("LBL".into()),
+                    span(2, 3, 2, 7)
+                )),
+                span(2, 3, 2, 7)
+            )]
+        );
```

Comment:
Testing errors here is a bitch, so I just added a label so that these are still passing.